internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2651,3 +2704,66 @@ func flattenKubernetesClusterMaintenanceConfigurationTimeInWeeks(input *[]contai
 	}
 	return results
 }
+
+func expandKubernetesClusterHttpProxyConfig(input []interface{}) *containerservice.ManagedClusterHTTPProxyConfig {
+	httpProxyConfig := containerservice.ManagedClusterHTTPProxyConfig{}
+	if len(input) == 0 {
+		return nil
+	}
+
+	config := input[0].(map[string]interface{})
+	httpProxyConfig.HTTPProxy = utils.String(config["http_proxy"].(string))
+	httpProxyConfig.HTTPSProxy = utils.String(config["https_proxy"].(string))
+	httpProxyConfig.TrustedCa = utils.String(config["trusted_ca"].(string))
+	noProxyRaw := config["no_proxy"].([]interface{})
+	httpProxyConfig.NoProxy = utils.ExpandStringSlice(noProxyRaw)
+	return &httpProxyConfig
```

Comment:
Whitespaces make it more readable in this case
```suggestion
	config := input[0].(map[string]interface{})

	httpProxyConfig.HTTPProxy = utils.String(config["http_proxy"].(string))
	httpProxyConfig.HTTPSProxy = utils.String(config["https_proxy"].(string))
	httpProxyConfig.TrustedCa = utils.String(config["trusted_ca"].(string))

	noProxyRaw := config["no_proxy"].([]interface{})
	httpProxyConfig.NoProxy = utils.ExpandStringSlice(noProxyRaw)

	return &httpProxyConfig
```